File: python/functions/class_example.py
#!/usr/bin/env python3

class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def greet(self):
        return f"Hello, I'm {self.name} and I'm {self.age} years old"

    def birthday(self):
        self.age += 1
        return f"Happy birthday! Now {self.age} years old"

    def __str__(self):
        return f"Person(name={self.name}, age={self.age})"


class Employee(Person):
    def __init__(self, name, age, job_title):
        super().__init__(name, age)
        self.job_title = job_title

    def work(self):
        return f"{self.name} is working as a {self.job_title}"

    def __str__(self):
        return f"Employee(name={self.name}, age={self.age}, job={self.job_title})"


person = Person("Alice", 30)
print(person.greet())
print(person.birthday())
print(person)

employee = Employee("Bob", 25, "Software Engineer")
print(employee.greet())
print(employee.work())
print(employee)

File: swift/functions/class_example.swift
#!/usr/bin/env swift

class Person {
    var name: String
    var age: Int

    init(name: String, age: Int) {
        self.name = name
        self.age = age
    }

    func greet() -> String {
        return "Hello, I'm \(name) and I'm \(age) years old"
    }

    func birthday() {
        age += 1
        print("Happy birthday! Now \(age) years old")
    }

    func description() -> String {
        return "Person(name=\(name), age=\(age))"
    }
}

class Employee: Person {
    var jobTitle: String

    init(name: String, age: Int, jobTitle: String) {
        self.jobTitle = jobTitle
        super.init(name: name, age: age)
    }

    func work() -> String {
        return "\(name) is working as a \(jobTitle)"
    }

    override func description() -> String {
        return "Employee(name=\(name), age=\(age), job=\(jobTitle))"
    }
}

let person = Person(name: "Alice", age: 30)
print(person.greet())
person.birthday()
print(person.description())

print("")

let employee = Employee(name: "Bob", age: 25, jobTitle: "Software Engineer")
print(employee.greet())
print(employee.work())
print(employee.description())
